src/sync.ts

```diff
@@ -1254,7 +1356,6 @@ export class SyncApi {
                 }
 
                 if (limited) {
-                    this.deregisterStateListeners(room);
```

Comment:
Removing the deregister and register here because it can now be handled automatically in one spot by `RoomEvent.CurrentStateUpdated` (see above)